### PR TITLE
fix: Correct bitbucket Author, Org, and Repo.

### PR DESCRIPTION
**Pull request by johnmshields**
The issue in jenkins-x/jx#664 is caused by some mismatching of the PR's
Author and the "username" in the various Bitbucket APIs. In the case
where the repository is part of an organization the "username" is
actually the organization name. Likewise, if the PR author does not
match the repository username the same issue would occur.

This commit updates the `bitbucket_cloud.go` provider to use the
organization name as the PR `Owner` and the short repository name as the
`Repo`. It also adds the `Author` to the returned PR and uses the
proviers `UserInfo` method to obtain the full author information.

I'll add that I don't have access to a Bitbucket Server instance at the moment to I can't confirm that these changes are needed there as well. My guess is that they are similar. I'd be happy to add changes to `bitbucket_server.go` if someone can confirm they are needed and/or test them.

### CODE REVIEW

**Comment by wbrefvem** [comment]
@jstrachan I tested the changes locally, and at this point, IMO, the failures are due to regressions. However, @johnmshields I'd love to hear your thoughts, and in any case thanks for the PR!

**Comment by wbrefvem** [comment]
Here's what I'm getting when I run `make test`:

```--- FAIL: TestBitbucketCloudProviderTestSuite (60.06s)
    --- FAIL: TestBitbucketCloudProviderTestSuite/TestListCommitStatus (0.00s)
    	require.go:765:
    			Error Trace:	bitbucket_cloud_test.go:312
    			            				bitbucket_cloud_test.go:332
    			Error:      	Expected nil, but got: &errors.errorString{s:"Status: 404 Not Found, Body: 404 page not found\n"}
    			Test:       	TestBitbucketCloudProviderTestSuite/TestListCommitStatus
    --- FAIL: TestBitbucketCloudProviderTestSuite/TestMergePullRequest (0.00s)
    	require.go:765:
    			Error Trace:	bitbucket_cloud_test.go:344
    			Error:      	Expected nil, but got: &errors.errorString{s:"Status: 404 Not Found, Body: 404 page not found\n"}
    			Test:       	TestBitbucketCloudProviderTestSuite/TestMergePullRequest
    --- FAIL: TestBitbucketCloudProviderTestSuite/TestPullRequestLastCommitStatus (0.00s)
    	require.go:765:
    			Error Trace:	bitbucket_cloud_test.go:306
    			Error:      	Expected nil, but got: &errors.errorString{s:"Status: 404 Not Found, Body: 404 page not found\n"}
    			Test:       	TestBitbucketCloudProviderTestSuite/TestPullRequestLastCommitStatus
```

**Comment by johnmshields** [comment]
Yes, let me take a look. I tried running last night with the clean master from jenkins-x/jx and got similar errors. I wanted to see what happened on the CI server... But maybe my environment wasn't fully clean. I'll take a look and let you know.

File: pkg/gits/bitbucket_cloud.go [hunk 1]
```diff
@@ -366,8 +367,8 @@ func (b *BitbucketCloudProvider) UpdatePullRequestStatus(pr *GitPullRequest) err
 	prID := int32(*pr.Number)
 	bitbucketPR, _, err := b.Client.PullrequestsApi.RepositoriesUsernameRepoSlugPullrequestsPullRequestIdGet(
 		b.Context,
-		b.Username,
-		strings.TrimPrefix(pr.Repo, pr.Owner+"/"),
```
Comment (wbrefvem) [inline pkg/gits/bitbucket_cloud.go]: The logic behind this and other instances of the same is that there were cases where a the `repo` argument was of the form `owner/repo` and it was necessary to trim the `owner/` portion. In any case, `strings.TrimPrefix` is idempotent and so this shouldn't need to be change (and doing so may produce regressions).

Comment (johnmshields) [reply pkg/gits/bitbucket_cloud.go]: I can re-review it, but what I found was that the reason it was coming in as `owner/repo` was due to the original PR creation logic that used `FullName` as the `pr.Repo` value. The updated code just uses the `Name` value to set the `pr.Repo`.

Everywhere I could find that this method was used it was getting the value from the `pr.Repo`.

Comment (wbrefvem) [reply pkg/gits/bitbucket_cloud.go]: Ah okay, that makes sense.

**Pull request merged**